libraries/src/main/java/com/baeldung/smack/StanzaThread.java

```diff
@@ -0,0 +1,57 @@
+package com.baeldung.smack;
+
+import org.jivesoftware.smack.AbstractXMPPConnection;
+import org.jivesoftware.smack.SmackException;
+import org.jivesoftware.smack.XMPPException;
+import org.jivesoftware.smack.chat2.Chat;
+import org.jivesoftware.smack.chat2.ChatManager;
+import org.jivesoftware.smack.packet.Message;
+import org.jivesoftware.smack.tcp.XMPPTCPConnection;
+import org.jivesoftware.smack.tcp.XMPPTCPConnectionConfiguration;
+import org.jxmpp.jid.impl.JidCreate;
+import org.jxmpp.stringprep.XmppStringprepException;
+
+import java.io.IOException;
+
+public class StanzaThread extends Thread{
```

Comment:
Extending `Thread` is an antipattern - is it necessary in this case? If not, let's change this to implement `Runnable`.

Reply:
Also, let's be consistent with our whitespacing on the curly at the end of lines. Sometimes you do it without a space between the final token and the brace, e.g. `Thread{` and other times you have a space, e.g. `run() {`. Let's use a space.